next-frontend/src/components/competitions/TabCompetitors.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+import React from "react";
+import { Card, Text, Table, Center, Spinner } from "@chakra-ui/react";
+import EventIcon from "@/components/EventIcon";
+import CountryMap from "@/components/CountryMap";
+import { useEffect, useState } from "react";
+
+const ALL_EVENTS = [
+  "333",
+  "222",
+  "444",
+  "555",
+  "666",
+  "777",
+  "333bf",
+  "333fm",
+  "333oh",
+  "clock",
+  "minx",
+  "pyram",
+  "skewb",
+  "sq1",
+  "444bf",
+  "555bf",
+  "333mbf",
+];
+
+interface CompetitorData {
+  id: string;
+}
+const TabCompetitors: React.FC<CompetitorData> = ({ id }) => {
+  const [competitors, setCompetitors] = useState<Person[]>([]);
+  const [eventIds, setEventIds] = useState<string[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    fetch(
+      `https://www.worldcubeassociation.org/api/v0/competitions/${id}/wcif/public`,
```

Comment:
The OpenAPI clients have typed URL helpers.

Reply:
not for wcif/public yet. But I can add that